### ISSUE TITLE
[feature request] DC-ASGD (Delay Compensated Asynchronous Stochastic Gradient Descent)

### ISSUE THREAD

**Issue by subcomputes** [pytorch/pytorch#6083]
DC-ASGD is an optimizer described in this [paper](https://arxiv.org/abs/1609.08326)

Frameworks like [MXNET](https://mxnet.apache.org/api/python/optimization/optimization.html#mxnet.optimizer.DCASGD) have the feature, are there plans to implement this feature into PyTorch? 

If approved by community, I would like to start the process of a feature request. I'm happy to create a pull request and implement the code after discussion.

**Comment by Stonesjtu** [comment]
I'm also planning to write an ASGD framework for pytorch. I think the feature is easy to implement in pytorch, but hard to tune for best performance.

**Comment by subcomputes** [comment]
Is there a preferred benchmark for performance? @soumith wrote convnet benchmark, I assume this is  the preferred benchmark for pytorch optimizations?

https://github.com/soumith/convnet-benchmarks

**Comment by yyn19951228** [comment]
Did anyone start the implementation?
